pytest_archon/rule.py

```diff
@@ -1,121 +1,150 @@
 from __future__ import annotations
+from dataclasses import dataclass
+import re
 
-from fnmatch import fnmatch
+from fnmatch import fnmatchcase
 from types import ModuleType
 from typing import Iterable, Sequence
 
 from pytest_archon.collect import ImportMap, collect_imports, walk, walk_runtime, walk_toplevel
 from pytest_archon.failure import add_failure  # type: ignore[import]
 
 
-def archrule(name: str, comment: str | None = None) -> Rule:
+@dataclass
+class RulePattern:
+    is_regex: bool
+    pattern: str
+
+    def match(self, k: str):
+        if self.is_regex:
+            return re.search(self.pattern, k)
+        else:
+            return fnmatchcase(k, self.pattern)
+
+    def __str__(self):
+        if self.is_regex:
+            return f"regex /{self.pattern}/"
+        else:
+            return f"fn pattern /{self.pattern}/"
+
+
+def _as_rule_patterns(use_regex_global, use_regex, patterns):
+    use_regex_verdict = use_regex_global if use_regex is None else use_regex
+    return [RulePattern(is_regex=use_regex_verdict, pattern=p) for p in patterns]
```

Comment:
I like this: now the how (of matching) the pattern can be separated from the checking code.

However, i think it would be simpler to have 2 `RulePattern` types, instead of using a boolean switch everywhere.

* `GlobPattern`
* `RegexPattern`

Strings can then be mapped to `GlobPatterns`

```
archrule("", "")
 # create regex pattern
.should_import(regex("foo.*"))
# Create a glob pattern
.should_import(glob("patt*n"))
# Simple way to create a glob pattern
.should_import("patt*n")
# (Bonus) Simple way to create a regex pattern, maybe?
.should_import("/patt.*/")
```

It would also allow us to mix and match glob and regex (which is  probably not such a good idea...).
It would also allow for custom `RulePattern` types. Thingies we haven't thought about yet.

Reply:
Then I would go either for
* only set `use_regex` on global lvl (via `archrule()`) or
* use the "patt*n" & "/patt.*/" syntax

Reply:
For separate types: I would (for now) keep only `RulePattern` which is initialized with a pattern string. It figues then out by itself if it is a regex or a glob pattern.

Reply:
Only setting `use_regex` on `archrule()` level will work for me.

Reply:
done